Ferry.java

```diff
@@ -1,11 +1,54 @@
+import java.util.ArrayList;
+
 /**
  * A ferry provides a one-way journey between two islands. It
  * has a ferry code and information about both the source and
  * the destination island
  *
- * @author (your name)
+ * @author (20067052)
  * @version (a version number or a date)
  */
 public class Ferry {
+    private int ferry_ID;
+    private Island source_island;
+    private Island destination_island;
+    private ArrayList<Pass> pass_inFerry;
+
+    public  Ferry(int ferryCode,Island sourceIsland,Island destinationIsland ) {
+        ferry_ID=ferryCode;
+        source_island=sourceIsland;
+        destination_island=destinationIsland;
+        pass_inFerry=new ArrayList<Pass>();
+    }
+    public int getFerryCode(){return ferry_ID;}
+    public Island getSourceIsland(){return source_island;}
+    public Island getDestinationIsland(){return destination_island;}
+    public boolean CanPassTravel(Pass temp){return temp.getCredits() >= 3;}
+    public String ProcessPass(Pass pass,Island island1,Island island2){
+        if(pass.getLuxuryRating()< island2.islandRating() ||
```

Comment:
The whole style here looks kinda bad. Maybe reformat it so each or operator is on a separate line with its condition. Up to you.

Reply:
noted i did smth but i dont know if its to your liking or not